exercises/practice/grains/.approaches/bit-shifting/content.md

```diff
@@ -0,0 +1,78 @@
+# Bit-shifting
+
+```csharp
+using System;
+using System.Numerics; // for use with BigInteger
+
+public static class Grains
+{
+    public static ulong Square(int n)
+    {
+        switch (n)
+        {
+            case int num when num > 0 && num < 65: return (ulong)1 << (num - 1);
+            default: throw new ArgumentOutOfRangeException("n must be 1 through 64");
+        }
```

Comment:
While I like the switch, I find the below easier to read:

```suggestion
        if (num < 1 || num > 64)
            throw new ArgumentOutOfRangeException("n must be 1 through 64");

        return (ulong)1 << (num - 1);
```

Interestingly, you use this structure in the other approach. Was there any particular reason why?

BTW I'm not opposed to having a separate section on how a switch can also be used, as it is a nice approach.

Reply:
No particular reason. It's just how I rolled for this one, and it's a nice variety from the other approach. I'll leave it for now. How a switch works is pretty basic, and I link to the doc, so I'll not add a section for now.